crates/rethnet_evm_napi/src/transaction/pending.rs

```diff
@@ -82,7 +82,10 @@ impl PendingTransaction {
     > {
         match &*self.inner {
             rethnet_eth::transaction::SignedTransaction::Legacy(transaction) => {
-                LegacySignedTransaction::new(&env, transaction).map(Either3::A)
+                LegacySignedTransaction::from_legacy(&env, transaction).map(Either3::A)
+            }
+            rethnet_eth::transaction::SignedTransaction::EIP155(transaction) => {
+                LegacySignedTransaction::from_eip155(&env, transaction).map(Either3::A)
```

Comment:
I'm also confused here if returning `LegacySignedTransaction` is correct

Reply:
https://github.com/NomicFoundation/hardhat/pull/4298#discussion_r1295003527